### PR TITLE
Ensure outgoing spans show on the Request dashboard in Sentry

### PR DESCRIPTION
**Pull request by jamescrosswell**
Resolves https://github.com/getsentry/sentry-dotnet/issues/3334

Tested using both `Sentry.Samples.Console.Basic` and `Sentry.Samples.OpenTelemetry.Console`. There's sometimes a bit of a delay between transactions showing in the Performance dashboard and the relevant spans showing in the Requests dashboard, but they are showing as expected (with the correct domain name).

### CODE REVIEW

File: samples/Sentry.Samples.Console.Basic/Sentry.Samples.Console.Basic.csproj [hunk 1]
```diff
@@ -35,5 +35,8 @@
   <ItemGroup>
     <ProjectReference Include="..\..\src\Sentry\Sentry.csproj" />
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="System.Net.Http" Version="4.3.4" />
```
Comment (bruno-garcia) [inline samples/Sentry.Samples.Console.Basic/Sentry.Samples.Console.Basic.csproj:39]: Is this needed for modern .NET or just .NET Framework?

**Comment by bitsandfoxes** [comment]
This looks really great! Thanks!
I'm just hesitant to merge this as I'm not sure about the implications: Will this affect alerts or users' dashboards?
Let me check with the team.

File: src/Sentry/SentryHttpMessageHandler.cs [hunk 2]
```diff
@@ -69,6 +69,10 @@ internal SentryHttpMessageHandler(IHub? hub, SentryOptions? options, HttpMessage
             $"{method} {url}" // e.g. "GET https://example.com"
             );
         span?.SetExtra(OtelSemanticConventions.AttributeHttpRequestMethod, method);
+        if (!string.IsNullOrWhiteSpace(request.RequestUri?.Host))
+        {
+            span?.SetExtra(OtelSemanticConventions.AttributeServerAddress, request.RequestUri!.Host);
```
Comment (bruno-garcia) [inline src/Sentry/SentryHttpMessageHandler.cs:74]: How about this? This way we don't need the `!` (assumign this compiled)

```suggestion
        if (request.RequestUri is not null && span is not null && !string.IsNullOrWhiteSpace(request.RequestUri.Host))
        {
            span.SetExtra(OtelSemanticConventions.AttributeServerAddress, request.RequestUri.Host);
```

**Review by bruno-garcia — APPROVED** [review]

File: samples/Sentry.Samples.Console.Basic/Sentry.Samples.Console.Basic.csproj (hunk 1 above)
Comment (jamescrosswell) [reply samples/Sentry.Samples.Console.Basic/Sentry.Samples.Console.Basic.csproj:39]: You're right... I can add `Condition="'$(TargetFramework)' == 'net462'"` here.

**Review by bitsandfoxes — APPROVED** [review]

**Pull request merged**